### PR TITLE
correct use of the txids from fabric

### PR DESCRIPTION
**Pull request by mbwhite**
Fixes issue https://github.com/hyperledger/composer/issues/936

### CODE REVIEW

File: packages/composer-common/lib/util.js [hunk 1]
```diff
@@ -73,22 +73,25 @@ class Util {
      * @param {SecurityContext} securityContext - The user's security context
      * @param {string} functionName - The name of the function to call.
      * @param {string[]} args - The arguments to pass to the function being called.
+     * @param {string} txId - Transaction Id to use
```
Comment (sstone1) [inline packages/composer-common/lib/util.js]: This should be `options`

Comment (sstone1) [reply packages/composer-common/lib/util.js]: @mbwhite another y u no run `npm test`!

```
lerna ERR! execute /home/travis/build/hyperledger/composer/packages/composer-common/lib/util.js
lerna ERR! execute   71:5  error  Expected JSDoc for 'options' but found 'txId'  valid-jsdoc
```

File: packages/composer-common/lib/util.js [hunk 2]
```diff
@@ -73,22 +73,25 @@ class Util {
      * @param {SecurityContext} securityContext - The user's security context
      * @param {string} functionName - The name of the function to call.
      * @param {string[]} args - The arguments to pass to the function being called.
+     * @param {Object} options - Any options to use for the invocation
```
Comment (davidkel) [inline packages/composer-common/lib/util.js]: I think we try to provide details of the options that can be accepted, for example
```
     * @param {Object} deployOptions connector specific deployment options
     * @param {string} deployOptions.logLevel the level of logging for the composer runtime
     * @param {any} deployOptions.endorsementPolicy the endorsement policy (either a JSON string or Object) as defined by fabric node sdk
     * @param {String} deployOptions.endorsementPolicyFile the endorsement policy json file containing the endorsement policy
```

File: packages/composer-connector-hlfv1/lib/hlfconnection.js [hunk 3]
```diff
@@ -763,10 +763,11 @@ class HLFConnection extends Connection {
      * @param {SecurityContext} securityContext The participant's security context.
      * @param {string} functionName The name of the chaincode function to invoke.
      * @param {string[]} args The arguments to pass to the chaincode function.
+     * @param {object} options Options that might be requried to for this connection
```
Comment (davidkel) [inline packages/composer-connector-hlfv1/lib/hlfconnection.js]: same here about possible option values, plus the text for options doesn't quite read correctly

File: packages/composer-common/lib/util.js [hunk 4]
```diff
@@ -64,31 +64,34 @@ class Util {
...
+        return securityContext.getConnection().queryChainCode(securityContext, functionName, args);
     }
 
     /**
      * Submit an invoke request to the chain-code
      * @param {SecurityContext} securityContext - The user's security context
      * @param {string} functionName - The name of the function to call.
      * @param {string[]} args - The arguments to pass to the function being called.
+     * @param {Object} options - options to pass to the invoking chain code
+     * @param {Object} options.transactionId Transaction Id to use.
```
Comment (sstone1) [inline packages/composer-common/lib/util.js]: These are optional (according to the code) so the JSDoc comments need wrapping with `[` and `]`.

File: packages/composer-common/lib/connection.js [hunk 5]
```diff
@@ -507,9 +509,11 @@ class Connection extends EventEmitter {
      * @param {SecurityContext} securityContext The participant's security context.
      * @param {string} functionName The name of the chaincode function to invoke.
      * @param {string[]} args The arguments to pass to the chaincode function.
+     * @param {Object} options options for the invoking chain code
+     * @param {Object} options.transactionId Transaction Id to use.
```
Comment (sstone1) [inline packages/composer-common/lib/connection.js]: These are optional (according to the code) so the JSDoc comments need wrapping with `[` and `]`.

File: packages/composer-common/lib/connection.js [hunk 6]
```diff
@@ -481,12 +481,14 @@ class Connection extends EventEmitter {
      * @param {SecurityContext} securityContext The participant's security context.
      * @param {string} functionName The name of the chaincode function to invoke.
      * @param {string[]} args The arguments to pass to the chaincode function.
+     * @param {Object} options Options for the invoking chaing code to use
+     * @param {Object} options.transactionId Transaction Id to use.
```
Comment (sstone1) [inline packages/composer-common/lib/connection.js]: These are optional (according to the code) so the JSDoc comments need wrapping with `[` and `]`.

**Pull request merged**